README.md

```diff
@@ -896,6 +901,81 @@ following articles:
+```
+mvn clean package -Pdist
+docker build . -t emissary:latest
+```
+
+### Run Emissary with Docker
+Once the image is successfully built, the image should be in your list of local images. Run ```docker images``` and there
+should be an entry for REPOSITORY:emissary and TAG:latest:
+```
+[~]$ docker images
+   REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
+   emissary            latest              e740d5f23a79        44 hours ago        620MB
+   centos              7                   49f7960eb7e4        2 days ago          200MB
+```
+
+To run files through Emissary, we'll need to volume mount local directories into the running container. Let's create two local directories for 
+input/output to Emissary:
+```
+mkdir input
+mkdir output
+```
+
+Now that we have two target directories, we can use the ```-v``` option to mount them into the container. To start Emissary, run the sample command:
+```
+docker run -it --rm -v ${PWD}/input:/opt/emissary/target/data:Z -v ${PWD}/output:/opt/emissary/localoutput:Z --name emissary emissary
+```
+
+Once Emissary starts up, we should see a log line that says: "Started EmissaryServer at http://localhost:8001." We now can copy files into 
+```input/InputData/``` for Emissary to process. When the processing has finish, the files will be moved to ```input/DoneData```. All extracted 
+content can be found in the local ```output``` directory. Depending on your specific configuration, there should be output in the ```output/json``` 
+directory.
+
+To monitor Emissary in a container, we need to connect to the running container. If we want to run the agents command,
+we simply need to run the following command:
+```
+docker exec -it emissary ./emissary agents -p 8001 --mon
+```
+
+To see the environment settings:
+```
+docker exec -it emissary ./emissary env
+```
+
+If you want to connect to the web front-end for Emissary, some additional steps are need to configure jetty. We need to
+give the container a hostname and pass that onto Emissary from the command line. Sample command:
+```
+docker run -it --rm --name emissary --hostname emissary-001 -p 8001:8001 emissary server -a 2 -p 8001 -s http -h emissary-001
+```
+
+Then from a browser, go to http://localhost:8001/ to see the endpoints.
+
+### Cluster Mode using Docker Compose
+We can use a Docker compose file to simulate cluster mode. We'll start a feeder and two workers by default. To start the cluster, run the 
+sample docker-compose.yml file:
+```
+docker-compose -f contrib/docker/docker-compose.yml up
+```
+
+Once the cluster has started, we can manage input and output by looking at the volumes that were created by the compose file. The volume
+ location can be found in ```/var/lib/docker/volumes/```. Simply move a file into the input directory for Emissary to process it:
+```
+cp emissary-knight.png /var/lib/docker/volumes/docker_input/_data/InputData/
```

Comment:
I don't have a /var/lib/docker when doing this. Do I need to do something special or do I need to create this directory?

Reply:
It is permission controlled so you might need to change your perms

Reply:
I am not quite sure I understand. Do I need to change the permissions of my /var/lib directory? Is this another step we should add to the README?

Reply:
Are you running with sudo? Some of these commands do not work if not setup to run without sudo. This is specific to Docker setup, so it should be in the Docker docs. 

Updated the README to include that you need to run without sudo.

Reply:
After investigation, /var/lib/docker/volumes has root root 700 permissions. This prohibits me from executing the sudo command

I have setup my docker group and added myself (pretty sure I did this before running into this issue the first time). I looked into the post-installation instructions and I didn't see anything in direct regards to this path/issue.

What are your permissions for this directory?

Reply:
changed this to use docker cp